### PR TITLE
S20 chapel link

### PR DESCRIPTION
**Pull request by Matt-Ramos**

### CODE REVIEW

**Comment by Matt-Ramos** [comment]
Clicking "need more chapel credits?" button from attended events page now links to events with chapel credits only. The URL also changes its params based on the filter that's selected. Works with back button and direct link as well.

File: src/views/Events/index.js [hunk 1]
```diff
@@ -50,22 +50,108 @@ export default class Events extends Component {
...
+  componentDidMount() {
     this.loadEvents();
+    window.addEventListener('resize', this.resize);
   }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.resize);
+  }
+
+  async loadPrevious() {
```
Comment (aridospassos) [inline src/views/Events/index.js:73]: a comment here explaining whats going on on this particular part would be really helpful.

File: src/views/Events/index.js [hunk 2]
```diff
@@ -50,22 +50,108 @@ export default class Events extends Component {
...
+      this.setState({ filteredEvents: events, loading: false });
+
+      // If the include past filter is on, we get the events from the past
+      if (this.state.includePast) {
+        /* When togglePastEvents() runs, it changes the state of includePast to its opposite.
+         * So, when includePast is true, we set it to false so that togglePastEvents() will set it
+         * back to true
+         */
+        this.setState({ includePast: false });
+        this.togglePastEvents();
```
Comment (aridospassos) [inline src/views/Events/index.js:113]: Might be good to put the state change after the function call. I could see the re-rendering causing some problems because it technically re-renders before getting to the function call.

File: src/views/Events/index.js [hunk 3]
```diff
@@ -50,22 +50,108 @@ export default class Events extends Component {
...
+      const events = await gordonEvent.getFilteredEvents(this.state);
+      this.setState({ filteredEvents: events, loading: false });
+
+      // If the include past filter is on, we get the events from the past
+      if (this.state.includePast) {
+        /* When togglePastEvents() runs, it changes the state of includePast to its opposite.
+         * So, when includePast is true, we set it to false so that togglePastEvents() will set it
+         * back to true
+         */
+        this.setState({ includePast: false });
```
Comment (russtuck) [inline src/views/Events/index.js:112]: I don't think there is any guarantee that setState() here will take effect before the next line.  See:
https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous
https://reactjs.org/docs/react-component.html#other-apis-1

File: src/views/Events/index.js [hunk 4]
```diff
@@ -101,6 +188,7 @@ export default class Events extends Component {
       const events = gordonEvent.getFutureEvents(allEvents); //Filter out past events initially
       this.setState({ allEvents, events, loading: false, filteredEvents: events });
     }
+    this.loadPrevious();
```
Comment (russtuck) [inline src/views/Events/index.js:191]: Why is this added?  A comment explaining the reasoning would help.

File: src/views/Events/index.js [hunk 5]
```diff
@@ -30,7 +30,7 @@ export default class Events extends Component {
       search: '',
       chapelCredits: false,
       art: false,
-      sports: false,
+      athletics: false,
```
[the remark this answers is not in the document]
Comment (Matt-Ramos) [reply src/views/Events/index.js]: good catch, fixed. reverted back to sports as it wasn't working

File: src/views/Events/index.js (hunk 2 above)
Comment (Matt-Ramos) [reply src/views/Events/index.js:129]: When togglePastEvents() runs, it changes the state of includePast to its opposite. So, when includePast is true, we set it to false so that togglePastEvents() will set it back to true.

If I put it after the function call then if i had the include past filter selected and refreshed the page, it would go away.

File: src/views/Events/index.js (hunk 1 above)
Comment (Matt-Ramos) [reply src/views/Events/index.js:74]: added comment

File: src/views/Events/index.js [hunk 6]
```diff
@@ -50,22 +50,108 @@ export default class Events extends Component {
...
+        athletics,
+        calendar,
+        cec,
+        fair,
+        chapelOffice,
+        studentLife,
+      });
+
+      this.setState({ loading: true });
+      const events = await gordonEvent.getFilteredEvents(this.state);
```
[the remark this answers is not in the document]
Comment (Matt-Ramos) [reply src/views/Events/index.js:119]: if I don't pass in the states, which are the filters in this case, it wont correctly filter

File: src/views/Events/index.js (hunk 3 above)
Comment (Matt-Ramos) [reply src/views/Events/index.js:128]: If I remove the setState or move it after the toggle, it doesn't work appropriately

File: src/views/Events/index.js (hunk 4 above)
Comment (Matt-Ramos) [reply src/views/Events/index.js:250]: added comment explaining that this was added to load previously set filters when loading the events either as an authenticated user or as a guest

File: src/views/Events/index.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (russtuck) [reply src/views/Events/index.js:117]: OK.  The state is not much more than the filters (just hasFilters and loading), so it's reasonable to pass the whole state.

File: src/views/Events/index.js (hunk 3 above)
Comment (russtuck) [reply src/views/Events/index.js:126]: I understand it won't work without it.  My concern is that in might sometimes not work even with it.

Comment (russtuck) [reply src/views/Events/index.js:126]: It's not save to call togglePastEvents() here.  I think you'll need to call just part of it.  See my note there about how to refactor.

File: src/views/Events/index.js [hunk 7]
```diff
@@ -79,6 +227,7 @@ export default class Events extends Component {
       const futureEvents = gordonEvent.getFutureEvents(this.state.allEvents);
       await this.setState({ events: futureEvents });
     }
+    this.createURLParameters();
```
Comment (russtuck) [inline src/views/Events/index.js:230]: I'd suggest breaking togglePastEvents() into 2 functions.  One is the toggle part, and the other is the rest, starting here.  Then you can call both, or just one.

File: src/views/Events/index.js [hunk 8]
```diff
@@ -123,8 +123,9 @@ export default class Events extends Component {
          * So, when includePast is true, we set it to false so that togglePastEvents() will set it
          * back to true
          */
-        this.setState({ includePast: false });
-        this.togglePastEvents();
+        this.setState({ includePast: false }, () => {
+          this.togglePastEvents();
```
Comment (russtuck) [inline src/views/Events/index.js:127]: This code is OK, but it's creating a new unnamed function that just calls a named function.  It would be slightly simpler and cleaner to say
  this.setState({ includePast: false }, this.togglePastEvents);
